Library/Homebrew/cmd/config.rb

```diff
@@ -119,8 +119,10 @@ def describe_system_ruby
     s
   end
 
-  def java_version
-    java = `java -version 2>&1`.lines.first.chomp
+  def describe_java
+    java = which "java"
+    return "N/A" if java.nil?
+    java = `#{java} -version 2>&1`.lines.first.chomp
```

Comment:
Yeh, that's a problem, annoying. What about just `/usr/libexec/java_home` alone?